### DISCUSSION TITLE
Component vs IComponent

### DISCUSSION THREAD

**Discussion by bykka** [richard-scryber/scryber.core#109] [Ideas]
Hi,

For composing a complex document we have to use **Component** instances (which implement **IComponent** interface) because **Page.Contents** property requires exactly **Component** instances .
Together with that **Document.ParseHtml** returns **IComponent** and that is not suitable for **Page.Contents**
So, I can't implement some components as html templates because have a problem to attache those to the Page.

Have a suggestion to change the result type of the method **Document.ParseHtml** from IComponent to Component. That should not brake code that already exist and will allow to mix different approaches for building document.

**Comment by richard-scryber** [comment]
Hi Pavlo, there are quite a few classes that implement the IComponent interface, that can be parsed, which are not actually inheriting from Scryber.Components.Component (styles, data classes etc). Along with any custom components others may have implemented.

A static method that returns a cast concrete instance should be fairly trivial to implement for anyone using the library. 

I have also posted a solution to your previous, excellent, discussion on binding SVG content, and I hope this provides a solution that alleviates this option as a request.

What do you think?

**Reply by bykka** [reply]
I personally had a problem to understand what method I can use and with what parameters
because depending on the value of the ParseSourceType parameter method has different behavior and it is not clean when it is possible to cast to Component class.

For example if I have the next fragment:
`string htmlFragment = "<h1>Table of contents</h1>"`

How to create a Component for that? Intuition says there should be a function (string htmlFragment) => Component
Everything else will look like work around.
